clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -218,4 +243,66 @@ public boolean seekToNewSource(long l) throws IOException {
             return false;
         }
     }
+
+    /**
+     * Returns Location with repository, ref and path used by lakeFS based on filesystem path
+     * @param path
+     * @return
+     */
+    private Location pathToLocation(Path path) {
+        if (!path.isAbsolute()) {
+            path = new Path(this.workingDirectory, path);
+        }
+
+        URI uri = path.toUri();
+        if (uri.getScheme() != null && uri.getPath().isEmpty()) {
+            return null;
+        }
+
+        Location loc = new Location();
+        loc.setRepository(uri.getHost());
+        // extract ref and rest of the path after removing the '/' prefix
+        String s = uri.getPath();
+        if (s.startsWith(SEPARATOR)) {
+            s = s.substring(1);
+        }
+        int i = s.indexOf(SEPARATOR);
+        if (i == -1) {
+            loc.setRef(s);
+        } else {
+            loc.setRef(s.substring(0, i));
+            loc.setPath(s.substring(i+1));
+        }
+        return loc;
+    }
+
+    private static class Location {
```

Comment:
Can you give it a name that will make it clear that the location is a lakefs location? maybe LakefsAddress / LakefsLocation
what do you think?

Reply:
Can rename to something else, but I think prefix Lakefs for everything in this package can be wrong.
Have other suggestions?

Reply:
yes, I hear you.. as discussed let's call it ObjectLocation for now